packages/frontend/src/components/LightdashVisualization/VisualizationCustomConfig.tsx

```diff
@@ -0,0 +1,55 @@
+import { ChartType, ItemsMap } from '@lightdash/common';
+import { FC, useEffect } from 'react';
+import useCustomVisualizationConfig from '../../hooks/useCustomVisualizationConfig';
+import {
+    VisualizationConfig,
+    VisualizationConfigCommon,
+} from './VisualizationProvider';
+
+export type VisualizationCustomConfigType = {
+    chartType: ChartType.CUSTOM;
+    chartConfig: ReturnType<typeof useCustomVisualizationConfig>;
+};
+
+export const isCustomVisualizationConfig = (
+    visualizationConfig: VisualizationConfig | undefined,
+): visualizationConfig is VisualizationCustomConfigType => {
+    return visualizationConfig?.chartType === ChartType.CUSTOM;
+};
+
+type VisualizationCustomConfigProps =
+    VisualizationConfigCommon<VisualizationCustomConfigType> & {
+        // TODO: shared prop once all visualizations are converted
```

Comment:
do we use this in custom viz?

Reply:
No, but I thought it was a good idea to keep it around cause that comment implies we want to use it everywhere.